portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -7097,6 +7126,44 @@ private boolean _isPasswordUnchanged(
 		return newEncPwd.equals(user.getPassword());
 	}
 
+	private void _onUserDoesNotExist(
+		String authType, long companyId, String login,
+		Map<String, String[]> headerMap) {
+
+		try {
+			AuditMessage auditMessage = null;
+
+			if (authType.equals(CompanyConstants.AUTH_TYPE_EA)) {
+				auditMessage = _buildAuditMessageUserDoesNotExist(
+					CompanyConstants.AUTH_TYPE_EA, login, companyId, headerMap);
+			}
+			else if (authType.equals(CompanyConstants.AUTH_TYPE_SN)) {
+				auditMessage = _buildAuditMessageUserDoesNotExist(
+					CompanyConstants.AUTH_TYPE_SN, login, companyId, headerMap);
+			}
+			else if (authType.equals(CompanyConstants.AUTH_TYPE_ID)) {
+				auditMessage = _buildAuditMessageUserDoesNotExist(
+					CompanyConstants.AUTH_TYPE_ID, login, companyId, headerMap);
+			}
```

Comment:
As these 3 get `auditMessage` the exact same way just passing in different strings, I think you should merge them and sort conditions alphabetically like:
```
if (authType.equals(CompanyConstants.AUTH_TYPE_EA) || authType.equals(CompanyConstants.AUTH_TYPE_ID) || authType.equals(CompanyConstants.AUTH_TYPE_SN)) {
	auditMessage = _buildAuditMessageUserDoesNotExist(
		authType, login, companyId, headerMap);
}
```

Reply:
Ok!
Then the `_buildAuditMessageUserDoesNotExist(...)` should be inlined as well because it will only be called once, right?

Reply:
Yeah, makes sense.